### PR TITLE
Fix datasource info view end time and duration

### PR DESCRIPTION
**Pull request by foxymiles**
**User-Facing Changes**
This fixes an issue with the display of end time and duration for non-live data sources.

**Description**
This fixes a regression from https://github.com/foxglove/studio/pull/4238. The issue is that non-live data sources don't send multiple updates for the session end time so for these data sources we also need to update when the player presence changes in order to display the correct end time and duration.

Fixes https://github.com/foxglove/studio/issues/4269

### CODE REVIEW

File: packages/studio-base/src/components/DataSourceInfoView.tsx [hunk 1]
```diff
@@ -74,7 +87,7 @@ function DataSourceInfoContent(props: {
         {playerPresence === PlayerPresence.INITIALIZING ? (
           <Skeleton animation="wave" width="50%" />
         ) : (
-          <Typography variant="inherit" ref={endTimeRef}>
+          <Typography className={classes.numericValue} variant="inherit" ref={endTimeRef}>
```
Comment (snosenzo) [inline packages/studio-base/src/components/DataSourceInfoView.tsx:90]: doesn't seem related to this issue but how is this updated? why doesn't it have a  `<Timestamp time={endTime}...` tag?

File: packages/studio-base/src/components/DataSourceInfoView.tsx [hunk 2]
```diff
@@ -16,9 +17,16 @@ import { useAppTimeFormat } from "@foxglove/studio-base/hooks";
 import { subtractTimes } from "@foxglove/studio-base/players/UserNodePlayer/nodeTransformerWorker/typescript/userUtils/time";
 import { PlayerPresence } from "@foxglove/studio-base/players/types";
 import { formatDate, formatDuration } from "@foxglove/studio-base/util/formatTime";
+import { fonts } from "@foxglove/studio-base/util/sharedStyleConstants";
 
 import { MultilineMiddleTruncate } from "./MultilineMiddleTruncate";
 
+const useStyles = makeStyles()({
+  numericValue: {
+    fontFamily: fonts.MONOSPACE,
```
Comment (defunctzombie) [inline packages/studio-base/src/components/DataSourceInfoView.tsx:26]: Can we get this from the mui theme instead? I think @2metres was trying to move away from referencing the sharedStyleConstants directly.

**Review by defunctzombie — APPROVED** [review]

File: packages/studio-base/src/components/DataSourceInfoView.tsx (hunk 1 above)
Comment (snosenzo) [reply packages/studio-base/src/components/DataSourceInfoView.tsx:90]: oh I see in the parent component that it's updating the DOM directly -- and I gather it's because the end time for live datasources is constantly changing 👍

File: packages/studio-base/src/components/DataSourceInfoView.tsx (hunk 2 above)
Comment (foxymiles) [reply packages/studio-base/src/components/DataSourceInfoView.tsx:26]: Yeah probably. I'd like to get this fix out since it's a release blocker but it seems like we shouldn't need to reference fonts this way.

File: packages/studio-base/src/components/DataSourceInfoView.tsx (hunk 1 above)
Comment (foxymiles) [reply packages/studio-base/src/components/DataSourceInfoView.tsx:90]: Yeah this is a performance hack so we can render this fast.
